tools/ports/__init__.py

```diff
@@ -27,6 +27,8 @@ def read_ports():
       assert hasattr(port, a), 'port %s is missing %s' % (port, a)
     if not hasattr(port, 'process_dependencies'):
       port.process_dependencies = lambda x: 0
+    if not hasattr(port, 'linker_setup'):
+      port.linker_setup = lambda x, y: 0
```

Comment:
is this related to the PR?

Reply:
Yes.  Previously ports were setting link-time flags as part of `process_dependencies`.. which happens during compile time.

This gives them second hook they can use for setting link time flags.   Without this change we would get assertion errors when the ports tries to set link-time flags during the compile phase.

Reply:
You can see it was just couple of ports that were messing with linker settings.   This change means they won't mess with those settings in `-c` calls..  making `-c` calls simpler and faster.